trust-quorum/src/task.rs

```diff
@@ -93,6 +198,114 @@ impl NodeTaskHandle {
         &self.baseboard_id
     }
 
+    /// Initiate a trust quorum reconfiguration at this node
+    pub async fn reconfigure(
+        &self,
+        msg: ReconfigureMsg,
+    ) -> Result<(), NodeApiError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .send(NodeApiRequest::Reconfigure { msg, responder: tx })
+            .await?;
+        rx.await??;
+        Ok(())
+    }
+
+    /// Initiate an LRTQ upgrade at this node
+    pub async fn upgrade_from_lrtq(
+        &self,
+        msg: LrtqUpgradeMsg,
+    ) -> Result<(), NodeApiError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .send(NodeApiRequest::LrtqUpgrade { msg, responder: tx })
+            .await?;
+        rx.await??;
+        Ok(())
+    }
+
+    /// Return the status of this node if it is coordinating the `Prepare` phase
+    /// of a reconfiguration or LRTQ upgrade. Return `Ok(None)` or an error
+    /// otherwise.
+    pub async fn coordinator_status(
+        &self,
+    ) -> Result<Option<CoordinatorStatus>, NodeApiError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .send(NodeApiRequest::CoordinatorStatus { responder: tx })
+            .await?;
+        let res = rx.await?;
+        Ok(res)
+    }
+
+    /// Load the rack secret for the given epoch
+    ///
+    /// This can block for an indefinite period of time before returning
+    /// and depends on availability of the trust quorum.
+    pub async fn load_rack_secret(
```

Comment:
Do we want a `try_load_rack_secret` that only tries once? Also should this log on retry? I'm a little worried this will be a bit of a black box that'll be hard to inspect in case it takes too long.

Reply:
My intention was that the caller would wrap this in their own timeout, rather than passing one in or retrying. However, it's different enough from everything else that this probably isn't warranted. I actually forgot about it at one point when writing tests and caused a hang. I should probably just not retry internally and return the `Result<Option<ReconstructedRackSecret>>`

Reply:
I think that would make sense.